node/src/cli/commands/login.command.ts

```diff
@@ -244,18 +253,132 @@ export class LoginCommand {
                     ' through the web vault to set your master password.');
             }
 
-            if (this.success != null) {
-                const res = await this.success();
-                return Response.success(res);
-            } else {
-                const res = new MessageResponse('You are logged in!', null);
-                return Response.success(res);
+            if (response.forcePasswordReset) {
+                return await this.updateTempPassword();
             }
+
+            return await this.handleSuccessResponse();
         } catch (e) {
             return Response.error(e);
         }
     }
 
+    private async handleSuccessResponse(): Promise<Response> {
+        if (this.success != null) {
+            const res = await this.success();
+            return Response.success(res);
+        } else {
+            const res = new MessageResponse('You are logged in!', null);
+            return Response.success(res);
+        }
+    }
+
+    private async updateTempPassword(error?: string): Promise<Response> {
+        // If no interaction available, alert user to use web vault
+        if (!this.canInteract) {
+            await this.logout();
+            this.authService.logOut(() => { /* Do nothing */ });
+            return Response.error(new MessageResponse('An organization administrator recently changed your master password. In order to access the vault, you must update your master password now via the web vault. You have been logged out.', null));
+        }
+
+        if (this.email == null || this.email === 'undefined') {
+            this.email = await this.userService.getEmail();
+        }
+
+        // Get New Master Password
+        const baseMessage = 'An organization administrator recently changed your master password. In order to access the vault, you must update your master password now.\n' + 'Master password: ';
+        const firstMessage = error != null ? error + baseMessage : baseMessage;
+        const mp: inquirer.Answers = await inquirer.createPromptModule({ output: process.stderr })({
+            type: 'password',
+            name: 'password',
+            message: firstMessage,
+        });
+        const masterPassword = mp.password;
+
+        // Master Password Validation
+        if (masterPassword == null || masterPassword === '') {
+            return this.updateTempPassword('Master password is required.\n');
+        }
+
+        if (masterPassword.length < 8) {
+            return this.updateTempPassword('Master password must be at least 8 characters long.\n');
+        }
+
+        // Get New Master Password Re-type
+        const retype: inquirer.Answers = await inquirer.createPromptModule({ output: process.stderr })({
+            type: 'password',
+            name: 'password',
+            message: 'Re-type New Master password:',
+        });
+        const masterPasswordRetype = retype.password;
+
```

Comment:
maybe display password strength between prompts? that way a user can use that feedback to intentionally fail retype and use a different password